### PR TITLE
In std:rc, clarify the lack of mutability inside an Rc

### PR DESCRIPTION
**Pull request by federicomenaquintero**
Also, point to the example in Cell's docs for how to do it.

### CODE REVIEW

File: src/liballoc/rc.rs [hunk 1]
```diff
@@ -17,9 +17,11 @@
 //! pointer to the same value in the heap. When the last [`Rc`] pointer to a
 //! given value is destroyed, the pointed-to value is also destroyed.
 //!
-//! Shared references in Rust disallow mutation by default, and `Rc` is no
-//! exception. If you need to mutate through an [`Rc`], use [`Cell`] or
-//! [`RefCell`].
+//! Shared references in Rust disallow mutation by default, and `Rc`
+//! is no exception: you cannot obtain a mutable reference to
+//! something inside an `Rc`.  If you need mutability, put a [`Cell`]
```
Comment (GuillaumeGomez) [inline src/liballoc/rc.rs]: Extra whitespace before "if you need mutability".

File: src/liballoc/rc.rs [hunk 2]
```diff
@@ -17,9 +17,11 @@
 //! pointer to the same value in the heap. When the last [`Rc`] pointer to a
 //! given value is destroyed, the pointed-to value is also destroyed.
 //!
-//! Shared references in Rust disallow mutation by default, and `Rc` is no
-//! exception. If you need to mutate through an [`Rc`], use [`Cell`] or
-//! [`RefCell`].
+//! Shared references in Rust disallow mutation by default, and `Rc`
```
Comment (GuillaumeGomez) [inline src/liballoc/rc.rs]: `Rc` should be [`Rc`].

File: src/liballoc/rc.rs (hunk 1 above)
Comment (GuillaumeGomez) [inline src/liballoc/rc.rs]: `Rc` should be [`Rc`].

**Review by GuillaumeGomez — CHANGES_REQUESTED** [review]
Few nits.

**Comment by GuillaumeGomez** [comment]
Just a few nits and it'll be good!

**Comment by federicomenaquintero** [comment]
Pushed an extra commit to de-nitify.

File: src/liballoc/rc.rs [hunk 3]
```diff
@@ -214,6 +216,7 @@
 //! [upgrade]: struct.Weak.html#method.upgrade
 //! [`None`]: ../../std/option/enum.Option.html#variant.None
 //! [assoc]: ../../book/method-syntax.html#associated-functions
+//! [introducing-mutability]: ../../std/cell/index.html#introducing-mutability-inside-of-something-immutable
```
Comment (GuillaumeGomez) [inline src/liballoc/rc.rs]: This line is longer than 100 characters so tidy fails.

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Once your tidy issue has been fixed, please squash your commits.

**Comment by federicomenaquintero** [comment]
I've pushed a squashed version.  This also has a shortened link name so the line in the references doesn't exceed 100 chars :-P

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+ rollup

**Pull request merged**